home/import_assessment_pages.py

```diff
@@ -0,0 +1,361 @@
+            for key, value in row.items()
+            if value and key in class_fields
+        }
+        try:
+            return cls(
+                tags=deserialise_list(row.pop("tags", "")),
+                question=[row.pop("question")] if row.get("question") else "",
+                answers=deserialise_list(row.pop("answers", "")),
+                scores=deserialise_list(row.pop("scores", "")),
+                **row,  # type: ignore
+            )
+        except TypeError:
+            raise ImportAssessmentException(
+                "The import file is missing some required fields."
+            )
+
+
+def group_rows_by_page_id(rows: list[ContentRow]) -> list[ContentRow]:
+    grouped_rows = defaultdict(list)
+
+    for row in rows:
+        grouped_rows[row.slug].append(row)
+    output_rows = []
+    for _slug, grouped_rows_list in grouped_rows.items():
+        questions = []
+        for grouped_row in grouped_rows_list:
+            for _, question in enumerate(grouped_row.question):
+                answers = grouped_row.answers if grouped_row.answers else []
+                scores = grouped_row.scores if grouped_row.scores else []
+                error = grouped_row.error if grouped_row.error else []
+                question_entry = {
+                    "question": question,
+                    "answers": [
+                        {"answer": ans.strip(), "score": sc.strip()}
+                        for ans, sc in zip(answers, scores, strict=False)
+                    ],  # Strip whitespace
+                    "error": error,
+                }
+                questions.append(question_entry)
+
+        output_rows.append(
+            ContentRow(
+                slug=grouped_rows_list[0].slug,
+                title=grouped_rows_list[0].title,
+                tags=grouped_rows_list[0].tags,
+                locale=grouped_rows_list[0].locale,
+                high_result_page=grouped_rows_list[0].high_result_page,
+                high_inflection=grouped_rows_list[0].high_inflection,
+                medium_result_page=grouped_rows_list[0].medium_result_page,
+                medium_inflection=grouped_rows_list[0].medium_inflection,
+                low_result_page=grouped_rows_list[0].low_result_page,
+                low_inflection=grouped_rows_list[0].low_inflection,
+                generic_error=grouped_rows_list[0].generic_error,
+                question_count=str(len(questions)),
+                question=questions,
+                answers=[],
+                scores=[],
+                error="",
+            )
+        )
+
+    return output_rows
+
+
+def result_page_get_id_from_slug(slug: list[str]) -> int:
+    try:
+        page = ContentPage.objects.get(slug=slug)
+    except ObjectDoesNotExist:
+        raise ImportAssessmentException(
+            f"You are trying to add an assessment with slug {slug} that does not exist. Please create the {slug} page first."
```

Comment:
It's not the assessment's slug that does not exist, it's one of the result page's slugs that doesn't exist.

So maybe a better wording for this is `You are trying to add an assessment, where one of the result pages references the content page with slug {slug}, which does not exist. Please create the content page first`